packages/node/test/machine/integration/install-virtual.spec.ts

```diff
@@ -0,0 +1,353 @@
+  proxyAddress: string,
+  amount: BigNumber
+) {
+  await transferERC20Tokens(
+    proxyAddress,
+    erc20ContractAddress,
+    erc20Contract.abi,
+    amount
+  );
+}
+
+async function setupChannel(
+  proxyAddress: string,
+  tokenAmounts: BigNumber,
+  tokenAddress: string
+) {
+  return StateChannel.setupChannel(
+    network.FreeBalanceApp,
+    proxyAddress,
+    xpubs
+  ).setFreeBalance(
+    createFreeBalanceStateWithFundedTokenAmounts(
+      multisigOwnerKeys.map<string>(key => key.address),
+      tokenAmounts,
+      [tokenAddress]
+    )
+  );
+}
+
+function createTargetAppInstance(stateChannel: StateChannel) {
+  return new AppInstance(
+    stateChannel.multisigAddress,
+    stateChannel.multisigOwners,
+    /* default timeout */ 0,
+    /* appInterface */ {
+      addr: twoPartyFixedOutcomeAppDefinition.address,
+      stateEncoding: "uint256",
+      actionEncoding: undefined
+    },
+    true, // virtual
+    0, // app sequence number
+    2, // latest state
+    1, // latest versionNumber
+    0, // latest timeout
+    {
+      playerAddrs: [AddressZero, AddressZero],
+      amount: Zero
+    },
+    undefined
+  );
+}
+
+async function ssoc(
+  targetAppInstance: AppInstance,
+  stateChannel: StateChannel
+) {
+  const setStateCommitment = new SetStateCommitment(
+    network,
+    targetAppInstance.identity,
+    targetAppInstance.hashOfLatestState,
+    targetAppInstance.versionNumber,
+    targetAppInstance.timeout
+  );
+
+  const setStateTx = setStateCommitment.getSignedTransaction([
+    // TODO: Replace with k-th signing keys later
+    multisigOwnerKeys[0].signDigest(setStateCommitment.hashToSign()),
+    multisigOwnerKeys[1].signDigest(setStateCommitment.hashToSign())
+  ]);
+
+  await wallet.sendTransaction({
+    ...setStateTx,
+    gasLimit: SETSTATE_COMMITMENT_GAS
+  });
+
+  const setStateCommitmentForFreeBalance = new SetStateCommitment(
+    network,
+    stateChannel.freeBalance.identity,
+    stateChannel.freeBalance.hashOfLatestState,
+    stateChannel.freeBalance.versionNumber,
+    0 // make the timeout 0 so this ends without a challenge timeout
+  );
+
+  await wallet.sendTransaction({
+    ...setStateCommitmentForFreeBalance.getSignedTransaction([
+      multisigOwnerKeys[0].signDigest(
+        setStateCommitmentForFreeBalance.hashToSign()
+      ),
+      multisigOwnerKeys[1].signDigest(
+        setStateCommitmentForFreeBalance.hashToSign()
+      )
+    ]),
+    gasLimit: SETSTATE_COMMITMENT_GAS
+  });
+
+  await appRegistry.functions.setOutcome(
+    targetAppInstance.identity,
+    targetAppInstance.encodedLatestState
+  );
+
+  await appRegistry.functions.setOutcome(
+    stateChannel.freeBalance.identity,
+    stateChannel.freeBalance.encodedLatestState
+  );
+}
+
+describe("ERC20", () => {
+  it("returns the ERC20", async done => {
+    proxyFactory.once("ProxyCreation", async (proxyAddress: string) => {
+      await fundWithDolphinCoin(wallet, proxyAddress, parseEther("10"));
```

Comment:
Maybe `fundStateChannelWithDolphinCoin`

Reply:
disagree, that name is more ambiguous than this one, since we are not doing the funding protocol, but just sending some dolphincoin into an address